app/Pasteboard.m

```diff
@@ -0,0 +1,258 @@
+#include <string.h>
+#import <UIKit/UIKit.h>
+#include "fs/poll.h"
+#include "fs/dyndev.h"
+#include "kernel/errno.h"
+#include "debug.h"
+
+// Prepare for fd separation
+#define fd_priv(fd) fd->clipboard
+
+#if 1
+#define CLIP_DEBUG_PRINTK(fmt, ...) printk(fmt, ##__VA_ARGS__)
+#else
+#define CLIP_DEBUG_PRINTK(...)
+#endif
+
+#define INITIAL_WBUFFER_SIZE 1024
+
+static void inline dump_clipboard_state(const char* tag, struct fd *fd) {
+    CLIP_DEBUG_PRINTK("%s offset=%d pb_gen=%d len=%lu\n", tag, fd->offset, fd_priv(fd).generation, UIPasteboard.generalPasteboard.string.length);
+}
+
+
+// If pasteboard contents were changed since file was opened,
+// all read operations on in return error
+static int check_read_generation(struct fd* fd) {
+    UIPasteboard *pb = UIPasteboard.generalPasteboard;
+
+    uint64_t pb_gen = (uint64_t)pb.changeCount;
```

Comment:
I prefer putting a space after a cast, and putting a space before the * in a pointer type. This is somewhat my fault for not getting clang-format set up though.

Reply:
@tbodt agreed on space before `*`, noted about space after cast.